From 75050f7b6213c8c81e72dd8ae23944bd5e7ea818 Mon Sep 17 00:00:00 2001
From: Dmitry Muhomor <muhomor.dmitry@gmail.com>
Date: Fri, 15 Mar 2024 15:27:21 +0200
Subject: [PATCH 25/37] SystemUI: respect setting for use of DeviceControlsTile
 from lockscreen

Handling of Settings toggle is in handleClick(), which is not reached when locked unless
isAllowedWhenLocked returns true.
---
 .../src/com/android/systemui/qs/tiles/DeviceControlsTile.kt     | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/qs/tiles/DeviceControlsTile.kt b/packages/SystemUI/src/com/android/systemui/qs/tiles/DeviceControlsTile.kt
index bb175e2a8..15622b436 100644
--- a/packages/SystemUI/src/com/android/systemui/qs/tiles/DeviceControlsTile.kt
+++ b/packages/SystemUI/src/com/android/systemui/qs/tiles/DeviceControlsTile.kt
@@ -100,6 +100,8 @@ class DeviceControlsTile @Inject constructor(
         }
     }
 
+    override fun isAllowedWhenLocked(action: Action) = true
+
     override fun handleClick(view: View?) {
         if (state.state == Tile.STATE_UNAVAILABLE) {
             return
-- 
2.47.0

